src/client/components/vision/network.ts

```diff
@@ -9,10 +9,13 @@ import { RobotModel } from '../robot/model'
 import { CameraModel } from './camera/model'
 import { VisionRobotModel } from './model'
 import Image = message.input.Image
+import CompressedImage = message.output.CompressedImage
 
 export class VisionNetwork {
+
   constructor(private network: Network) {
     this.network.on(Image, this.onImage)
+    this.network.on(CompressedImage, this.onImage)
```

Comment:
Looks like the `image` param type on the `onImage` method should be `Image | CompressedImage`

Reply:
Or two separate methods that delegate.